app/controllers/partners/children_controller.rb

```diff
@@ -36,10 +36,7 @@ def new
       family = current_partner.families.find_by!(id: params[:family_id])
       @child = family.children.new
 
-      requestable_items = PartnerFetchRequestableItemsService.new(partner_id: current_partner.id).call
-      @formatted_requestable_items = requestable_items.map do |rt|
-        [rt.name, rt.id]
-      end
+      set_formated_requestable_items
```

Comment:
`formatted`

Reply:
Also, I'd prefer not having side effects from a method. So if the method could return the items, I'd like this line to be 

`@requestable_items = formatted_requestable_items`

Reply:
Also also... it looks like this service is only used in these controllers. So it might be simpler to just change the service to return the mapped name and ID, since that's all we care about. And then we can leave the line as

```
@requestable_items = PartnerFetchRequestableItemsService.new(partner_id: current_partner.id).call
```

I think the `formatted` part is overkill, I'd just rename the variable.

Reply:
I like this a lot. I didn't like my side effects but couldn't come up with a good way to avoid them.